orval-labs/orval
# fix: required of parent schema not considered in allOf #1570

## Status

**READY**

## Description

Fixes #1570

Given a schema like

````yaml
components:
  schemas:
    NewTicketNotification:
      required:
        - ctime
        - id
        - ticketId
        - type
        - userId
      type: object
      allOf:
        - $ref: '#/components/schemas/Notification'
        - type: object
          properties:
            ticketId:
              type: integer
              format: int32
            userId:
              type: integer
              format: int32
````

the `allOf` was considered as schema on its own when resolving the value of the fields, but in fact we want to consider the `required` fields of the parent schema also for the `allOf` sub schema.

This fixes this by including the `required` field of the parent schema in the sub schema.

## Related PRs

There is this existing outdated PR: https://github.com/orval-labs/orval/pull/861

## Todos


## Steps to Test or Reproduce

```bash
> git clone https://github.com/MGabr/orval.git
> yarn build
> cd tests
> yarn generate:react-query
```

The TypeScript model generated for the new `polymorphic-required.yaml` test spec should have the required fields set correctly. 

So for 

````yaml
components:
  schemas:
    DescendantOne:
      required:
        - type
        - count
      type: object
      allOf:
        - $ref: '#/components/schemas/ParentType'
        - type: object
          properties:
            value:
              type: boolean
            count:
              type: integer
              format: int32
            otherCount:
              type: integer
              format: int32
````

it generates

````typescript
export type DescendantOneAllOf = {
  count: number;
  otherCount?: number;
  value?: boolean;
};
````
